classes/Product.php

```diff
@@ -3276,11 +3276,16 @@ public static function getPricesDrop(
         $ids_product = Product::_getProductIdByDate((!$beginning ? $current_date : $beginning), (!$ending ? $current_date : $ending), $context);
 
         $tab_id_product = [];
+        
         foreach ($ids_product as $product) {
             if (is_array($product)) {
-                $tab_id_product[] = (int) $product['id_product'];
+                if(!in_array($product['id_product'], $tab_id_product)){
+                    $tab_id_product[] = (int) $product['id_product'];
+                }
```

Comment:
```suggestion
                $productId = (int) $product['id_product'];
                if(!in_array($productId, $tab_id_product)){
                    $tab_id_product[] = $productId;
                }
```
Otherwise, you are not testing the casted int in the in_array call

Reply:
good note!